### PR TITLE
Adds renovate config removes dependabot, increases codecov action version

### PR DESCRIPTION
**Pull request by mustafacantekir**
Fixes https://github.com/wayfair-incubator/little-cheesemonger/issues/131
Increases the codecov action version to fix the pipeline because [codecov action v1 is deprecated as of  February 1, 2022](https://github.com/codecov/codecov-action#%EF%B8%8F--deprecation-of-v1)

- [x]  I have read the Adding Renovate to Existing Projects guide.
- [x]  I have assigned this issue to myself avoid duplicating efforts with other potential contributors.
- [x]  I have verified this repository does not already have Renovate configured (or proposed in an open PR by another contributor).
- [x]  If the renovate[bot] account has already auto-filed a Configure Renovate PR in this repository, I confirm that I will create a separate PR under my own GitHub account, using the initial PR as inspiration.
- [x]  I confirm that I have added Renovate linting to this project's existing CI pipeline, or have created a new linting workflow if one doesn't already exist.
- [x]  If this repository is currently configured to use GitHub's Dependabot, my PR will also deprecate support for Dependabot in order to avoid conflicts with Renovate

### CODE REVIEW

File: .github/workflows/renovate_linting.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  renovate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: 🧼 lint renovate config # Validates changes to renovate.json config file
+        uses: suzuki-shunsuke/github-action-renovate-config-validator@v0.1.2
+        with:
+          config_file_path: 'renovate.json'
```
Comment (draco2003) [inline .github/workflows/renovate_linting.yml:12]: Minor nit: Add newline at end of file:
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.new_line_at_end_of_file

File: renovate.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  "packageRules": [
+    {
+      "matchPackageNames": "pip_requirements",
+      "matchFiles": [
+        "requirements.txt",
+        "requirements-test.txt"
+      ]
+    }
+  ]
+}
```
Comment (draco2003) [inline renovate.json:25]: Minor nit: Add newline at end of file:
https://eslint.org/docs/latest/rules/eol-last#options

**Review by chrisantonellis — APPROVED** [review]
LGTM pending comment about newline

File: .github/workflows/renovate_linting.yml (hunk 1 above)
Comment (mustafacantekir) [reply .github/workflows/renovate_linting.yml]: Added a new line for both files, doesn't show up on the diff though

**Pull request merged**